packages/compiler/src/render3/view/styling_builder.ts

```diff
@@ -377,22 +379,32 @@ export class StylingBuilder {
     // be evaluated (the AST visit call) during creation time so that any
     // pipes can be picked up in time before the template is built
     const mapValue = stylingInput.value.visit(valueConverter);
+    let reference: o.ExternalReference;
     if (mapValue instanceof Interpolation) {
+      if (!isClassBased) {
+        // We only support interpolations inside class bindings.
+        error('Unexpected interpolation');
+      }
+
       totalBindingSlotsRequired += mapValue.expressions.length;
+      reference = isClassBased ? getClassMapInterpolationExpression(mapValue) : R3.styleMap;
```

Comment:
The fact that we fall back to `R3.styleMap` here seems strange since we are throwing an error on line 386 if `!isClassBased`.  Seems like we can assume at this point that it's class-based, no?

```suggestion
      reference = getClassMapInterpolationExpression(mapValue);
```